servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/AbstractTimeoutHttpFilter.java

```diff
@@ -151,6 +158,62 @@ private Executor contextExecutor(final HttpRequestMetaData requestMetaData,
                 context.executor() : context.ioExecutor();
     }
 
+    // This is responsible for ensuring that we don't abandon the response resources due to the use of Single.timeout.
+    // It does so by retaining a reference to the StreamingHttpResponse in case we receive a cancellation. It will
+    // then attempt to drain the resource. We're protected from multiple subscribes because the TimeoutSingle will
+    // short circuit also on upstream cancellation, although that is not obviously correct behavior.
+    private static final class CleanupSubscriber implements SingleSource.Subscriber<StreamingHttpResponse> {
```

Comment:
However, it will still log that "this is a strong indication of a bug in a user-defined filter" and confuse users. Probably better to keep `CleanupSubscriber`